server/test/50-recipe.route.test.ts

```diff
@@ -2,20 +2,29 @@ process.env["NODE_ENV"] = "test";
 
 import { expect } from "chai";
 import * as mocha from "mocha";
+
```

Comment:
could you please reorder / group your tests. it costs many time if you override the token so much. :-)

Reply:
No, unfortunately not. The main problem is the the test don't reset the state of the db before each test. So each test depends on the test before it since the state of the db was changed. For example I had problems to use a specific user id in the recipes because it was the id you deleted your last user test (The cause of this error was a really hard to find). May I can check if I can reorder some of them but I also need a lot of different users to test the authorization stuff.

But in general it would be a good practice to reset the db after each test. What do you think?

Reply:
yes, i have the same big issues inside the fooditem tests. how could we realize the db restore after each test? i think call `npm test` with `require('child_process').exec()` is not the best solution!? maybe the restoring of the db after each test cost many time. what if we use the before or after hook?

Reply:
we could implement the db setup programmatically inside the test folder similar i did [here](https://github.com/flashback2k14/playthatsong/blob/master/server/logic/services/setup.service.js)!?

Reply:
Yes, we should do it this way and then use beforeEach and afterEach to do the setup and the cleanup?!

Reply:
And I think because these tests are integration tests the speed is not the most important aspect because they are already a lot slower then unit test (because they use http an db connections and so on)

Reply:
We should prefer stable and correct test over the execution time in this case I think.

Reply:
Yes, you are right. So think the beforeEach hook should create a fresh DB. AfterEach hook it's not needed than.